idea/src/org/jetbrains/kotlin/idea/inspections/RemoveRedundantSpreadOperatorInspection.kt

```diff
@@ -31,6 +35,16 @@ class RemoveRedundantSpreadOperatorInspection : AbstractKotlinInspection() {
                 when (argumentExpression) {
                     is KtCallExpression -> {
                         if (!argumentExpression.isArrayOfMethod()) return
+                        if (argumentExpression.valueArguments.isEmpty()) {
+                            val call = argument.getStrictParentOfType<KtCallExpression>()
+                            if (call != null) {
+                                val index = call.valueArguments.indexOfFirst { it == argument }
+                                val newCall = call.copy() as KtCallExpression
+                                newCall.valueArgumentList?.removeArgument(index)
+                                val newContext = newCall.analyzeAsReplacement(call, call.analyze(BodyResolveMode.PARTIAL))
+                                if (newCall.getResolvedCall(newContext)?.resultingDescriptor == null) return
```

Comment:
This check doesn't work for:
```kotlin
fun foo(vararg args: Int) = println(1)
fun foo() = println(2)
fun test() {
    foo(*intArrayOf())
} 
```
I think you should compare result descriptor with previous value.